### PR TITLE
[Private] Az.StorageSync module update to Shared/storage sync

### PR DESCRIPTION
**Pull request by ankushbindlish2**
## Description
- Record all tests
- Fix bug of Register server with assign identity
- Add clean keys for recorded tests
- Handle RoleAssignmentexists error came from RoleAssignment Create call
- Update Module Version to disambiguate with public version.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/StorageSync/StorageSync/StorageSyncService/NewStorageSyncServiceCommand.cs [hunk 1]
```diff
@@ -110,8 +110,8 @@ public class NewStorageSyncServiceCommand : StorageSyncClientCmdletBase
             Mandatory = false,
             HelpMessage = HelpMessages.StorageSyncServiceIdentityTypeParameter)]
         [ValidateSet(StorageSyncServiceIdentityType.systemAssigned,
-            StorageSyncServiceIdentityType.userAssigned,
-            StorageSyncServiceIdentityType.systemAssignedUserAssigned,
+            //StorageSyncServiceIdentityType.userAssigned,
```
Comment (VeryEarly) [inline src/StorageSync/StorageSync/StorageSyncService/NewStorageSyncServiceCommand.cs:113]: this is breaking change removing values from validate set

File: src/StorageSync/StorageSync/StorageSyncService/SetStorageSyncServiceCommand.cs [hunk 2]
```diff
@@ -123,8 +123,8 @@ public class SetStorageSyncServiceCommand : StorageSyncClientCmdletBase
             Mandatory = false,
             HelpMessage = HelpMessages.StorageSyncServiceIdentityTypeParameter)]
         [ValidateSet(StorageSyncServiceIdentityType.systemAssigned,
-            StorageSyncServiceIdentityType.userAssigned,
-            StorageSyncServiceIdentityType.systemAssignedUserAssigned,
+            //StorageSyncServiceIdentityType.userAssigned,
```
Comment (VeryEarly) [inline src/StorageSync/StorageSync/StorageSyncService/SetStorageSyncServiceCommand.cs:126]: this is breaking change removing values from validate set

**Review by VeryEarly — CHANGES_REQUESTED** [review]

File: src/StorageSync/StorageSync/Az.StorageSync.psd1 [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '2.3.1'
+ModuleVersion = '2.3.2'
```
[the remark this answers is not in the document]
Comment (ankushbindlish2) [reply src/StorageSync/StorageSync/Az.StorageSync.psd1:15]: Will revert once pushing to main.

File: src/StorageSync/StorageSync/StorageSyncService/NewStorageSyncServiceCommand.cs (hunk 1 above)
Comment (ankushbindlish2) [reply src/StorageSync/StorageSync/StorageSyncService/NewStorageSyncServiceCommand.cs:113]: Not really
- we never supported this value from service

File: src/StorageSync/StorageSync/StorageSyncService/SetStorageSyncServiceCommand.cs (hunk 2 above)
Comment (ankushbindlish2) [reply src/StorageSync/StorageSync/StorageSyncService/SetStorageSyncServiceCommand.cs:126]: We never supported this value

**Pull request merged**